### PR TITLE
#159951821 adding functionality to edit user profiles

### PR DESCRIPTION
**Pull request by phillipseryazi**
#### What does this PR do?
enables users to view and edit their profiles
#### Description of Task to be completed?
Users should be able to amend/edit their profiles depending on the change of preference to the full disclosure of the individual user
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/159951821
#### Test coverage 
screen shot 2018-09-12 at 22 01 20
screen shot 2018-09-13 at 12 21 13

### CODE REVIEW

**Comment by geofrocker** [comment]
Did you follow [this](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices)

File: package.json [hunk 1]
```diff
@@ -47,9 +49,9 @@
     "react-social-login-buttons": "^2.1.2",
     "react-test-renderer": "^16.4.2",
     "redux": "^4.0.0",
-    "redux-mock-store": "^1.5.3",
```
Comment (geofrocker) [inline package.json]: why did you remove this?

File: src/components/Profile/EditProfileModal.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    this.setState({
+      imageFile: event.target.files[0],
+    });
+  };
+
+  fileUploadHandler = () => {
+    const { imageFile } = this.state;
+    const formData = new FormData();
+    formData.append('file', imageFile);
+    const API_KEY = '156989293385565';
```
Comment (geofrocker) [inline src/components/Profile/EditProfileModal.js]: Keys should be stored as environment variables

File: src/components/Profile/EditProfileModal.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+          </div>
+        </div>
+      </Wrapper>
+    );
+  }
+}
+EditProfileModal.propTypes = {
+  bio: PropTypes.string.isRequired,
+  username: PropTypes.string.isRequired,
+  // image: PropTypes.string.isRequired,
```
Comment (geofrocker) [inline src/components/Profile/EditProfileModal.js]: remove commented code

**Comment by AGMETEOR** [comment]
The styling doesn't seem consistent with what we are using. The is no margin on the left and right of the body. Also, I receive articles I didn't create on my profile.

**Comment by AGMETEOR** [comment]
@Seryazi-Andela The landing page also doesn't load.

File: package.json (hunk 1 above)
Comment (phillipseryazi) [reply package.json]: I was getting errors from this

File: src/components/Profile/EditProfileModal.js (hunk 3 above)
Comment (phillipseryazi) [reply src/components/Profile/EditProfileModal.js]: Okay will do

File: src/components/Profile/EditProfileModal.js (hunk 2 above)
Comment (phillipseryazi) [reply src/components/Profile/EditProfileModal.js]: I have created the environment variables

File: public/index.html [hunk 4]
```diff
@@ -38,5 +39,12 @@
...
-</html>
+  <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo"
+    crossorigin="anonymous"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js" integrity="sha384-ZMP7rVo3mIykV+2+9J3UJ46jBk0WLaUAdn689aCwoqbBJiSnjAK/l8WvCWPIPm49"
+    crossorigin="anonymous"></script>
+  <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.min.js" integrity="sha384-ChfqqxuZUCnJSK3+MXmPNIyE6ZbWh2IMqE241rYiqJxyMiZ6OW/JmZQ5stwEULTy"
+    crossorigin="anonymous"></script>
+</body>
+
+</html>
```
Comment (geofrocker) [inline public/index.html]: EOF missing here

**Comment by lytes20** [comment]
@Seryazi-Andela , this is what is on your deployed branch, could you please correct this
screen shot 2018-09-12 at 10 12 27

File: package.json (hunk 1 above)
Comment (geofrocker) [reply package.json]: Some files depend on this

**Comment by geofrocker** [comment]
Add a screenshot of your tests

File: public/index.html [hunk 5]
```diff
@@ -41,5 +43,12 @@
...
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
-  </body>
+  <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo"
+    crossorigin="anonymous"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js" integrity="sha384-ZMP7rVo3mIykV+2+9J3UJ46jBk0WLaUAdn689aCwoqbBJiSnjAK/l8WvCWPIPm49"
+    crossorigin="anonymous"></script>
+  <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.min.js" integrity="sha384-ChfqqxuZUCnJSK3+MXmPNIyE6ZbWh2IMqE241rYiqJxyMiZ6OW/JmZQ5stwEULTy"
+    crossorigin="anonymous"></script>
+</body>
```
Comment (geofrocker) [inline public/index.html:52]: What are using this for?

Comment (phillipseryazi) [reply public/index.html:52]: It's for the pop up on the edit profile feature

Comment (geofrocker) [reply public/index.html:52]: This is not the first popup in this application, we have popups for login and registration

Comment (geofrocker) [reply public/index.html:52]: Having a sync with the people that handled those features will be helpful

Comment (phillipseryazi) [reply public/index.html:52]: what if there pops ups use different cdns

Comment (phillipseryazi) [reply public/index.html:52]: i can't remove it my pop up wont work

File: src/tests/containers/Profile.test.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+      ],
+    };
+    it('should call handle response', async () => {
+      moxios.stubRequest('https://authors-haven-tabs.herokuapp.com/api/articles/all/', {
+        status: 200,
+        response: { results: { articles } },
+      });
+      const spy = jest.spyOn(wrapper.instance(), 'handleResponse');
+      // wrapper.instance().forceUpdate();
+      // console.log(wrapper.instance());
```
Comment (abulojoshua1) [inline src/tests/containers/Profile.test.js]: @Seryazi-Andela, do you still need this commented out code? I think it will be better if it is not there.

File: src/routes.js [hunk 7]
```diff
@@ -21,6 +22,7 @@ const routes = (props) => {
           <Switch>
             {authStatus ? <Route path="/articles/:slug" exact component={Article} /> : null}
             {authStatus ? <Route path="/createArticle" exact component={CreateArticle} /> : null}
+            <Route path="/profile" exact component={Profile} />
```
Comment (abulojoshua1) [inline src/routes.js]: @Seryazi-Andela, I think the profile editing route should be protected and only viewable by authorized users.

File: src/tests/containers/Profile.test.js [hunk 8]
```diff
@@ -0,0 +1,133 @@
+import { mount } from 'enzyme';
+// import { shallow } from 'enzyme';
+// import { equal } from 'assert';
+import moxios from 'moxios';
+import React from 'react';
+// import { compose } from 'redux';
```
Comment (abulojoshua1) [inline src/tests/containers/Profile.test.js]: @Seryazi-Andela, I think the commented out imports will not be necessary here.

**Review by abulojoshua1 — CHANGES_REQUESTED** [review]

**Review by abulojoshua1 — APPROVED** [review]
